### PR TITLE
[ticket/17176] Upgrade to Symfony 6.3 and PHP 8.1+

### PR DESCRIPTION
**Pull request by marc1706**
Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/master/coding-guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket:

https://tracker.phpbb.com/browse/PHPBB3-17176

### CODE REVIEW

**Comment by rxu** [comment]
Tested and `dispatcher` => `event_dispatcher` compiler pass seems to work with extensions.

**Comment by rxu** [comment]
phpBB tests framework also has some workarounds to provide BC for phpUnit < 9.0 relied on `self::$phpunit_version = PHPUnit\Runner\Version::id();` As far as I could remind it's in `phpbb_test_case` and `phpbb_database_test_case`, probably somewhere else.
Those become outdated with this PR.

**Comment by marc1706** [comment]
> phpBB tests framework also has some workarounds to provide BC for phpUnit < 9.0 relied on `self::$phpunit_version = PHPUnit\Runner\Version::id();` As far as I could remind it's in `phpbb_test_case` and `phpbb_database_test_case`, probably somewhere else. Those become outdated with this PR.

Should be resolved by latest commit

File: phpBB/phpbb/request/deactivated_super_global.php [hunk 1]
```diff
@@ -103,7 +105,7 @@ public function offsetUnset($offset)
 	* @return void
 	* @psalm-suppress InvalidReturnType
 	*/
-	public function count()
+	public function count(): int
```
Comment (m-ober) [inline phpBB/phpbb/request/deactivated_super_global.php:106]: Why use `int` and not `void` together with `#[ReturnTypeWillChange]` like for `getIterator()`?

**Review by paul999 — APPROVED** [review]

File: phpBB/composer.json [hunk 2]
```diff
@@ -35,53 +35,53 @@
...
-		"symfony/http-kernel": "^5.4",
+		"symfony/dependency-injection": "^6.3",
+		"symfony/event-dispatcher": "^6.3",
+		"symfony/filesystem": "^6.3",
+		"symfony/finder": "^6.3",
+		"symfony/http-foundation": "^6.3",
+		"symfony/http-kernel": "^6.3",
 		"symfony/polyfill-intl-normalizer": "^1.23",
 		"symfony/polyfill-mbstring": "^1.23",
 		"symfony/polyfill-php72": "^1.23",
```
Comment (rubencm) [inline phpBB/composer.json:55]: Remove this

**Review by CHItA — APPROVED** [review]

**Comment by rxu** [comment]
Tested installing normal way, worked fine.

**Pull request merged**